repair_discount/models/mrp_repair.py

```diff
@@ -9,94 +9,50 @@
-                fee.price_unit,
-                fee.repair_id.pricelist_id.currency_id,
-                fee.product_uom_qty,
-                fee.product_id,
-                fee.repair_id.partner_id,
-            )
-            fee.price_subtotal = taxes["total_excluded"] * (
-                1 - (fee.discount or 0.0) / 100.0
-            )
+    discount = fields.Float(
+        string="Discount (%)",
+        store=True,
+    )
 
     @api.depends(
-        "price_unit", "repair_id", "product_uom_qty", "product_id", "tax_id", "discount"
+        "discount",
     )
-    def _compute_price_total(self):
+    def _compute_price_total_and_subtotal(self):
+        res = super()._compute_price_total_and_subtotal()
         for fee in self:
-            taxes = fee.tax_id.compute_all(
-                fee.price_unit,
-                fee.repair_id.pricelist_id.currency_id,
-                fee.product_uom_qty,
-                fee.product_id,
-                fee.repair_id.partner_id,
-            )
-            fee.price_total = taxes["total_included"] * (
+            fee.price_total = fee.price_total * (1 - (fee.discount or 0.0) / 100.0)
+            fee.price_subtotal = fee.price_subtotal * (
                 1 - (fee.discount or 0.0) / 100.0
             )
+        return res
 
 
 class RepairLine(models.Model):
     _inherit = "repair.line"
 
-    discount = fields.Float(string="Discount (%)")
-
-    @api.depends(
-        "price_unit",
-        "repair_id",
-        "product_uom_qty",
-        "product_id",
-        "repair_id.invoice_method",
-        "discount",
+    discount = fields.Float(
+        string="Discount (%)",
+        store=True,
     )
-    def _compute_price_subtotal(self):
-        for line in self:
-            taxes = self.env["account.tax"].compute_all(
-                line.price_unit,
-                line.repair_id.pricelist_id.currency_id,
-                line.product_uom_qty,
-                line.product_id,
-                line.repair_id.partner_id,
-            )
-            line.price_subtotal = taxes["total_excluded"] * (
-                1 - (line.discount or 0.0) / 100.0
-            )
 
     @api.depends(
-        "price_unit",
-        "repair_id",
-        "product_uom_qty",
-        "product_id",
-        "tax_id",
-        "repair_id.invoice_method",
         "discount",
     )
-    def _compute_price_total(self):
+    def _compute_price_total_and_subtotal(self):
+        res = super()._compute_price_total_and_subtotal()
         for line in self:
-            taxes = line.tax_id.compute_all(
-                line.price_unit,
-                line.repair_id.pricelist_id.currency_id,
-                line.product_uom_qty,
-                line.product_id,
-                line.repair_id.partner_id,
-            )
-            line.price_total = taxes["total_included"] * (
+            line.price_total = line.price_total * (1 - (line.discount or 0.0) / 100.0)
```

Comment:
Calculate `1 - line.discount / 100.0` once and use it for both. The `or 0.0` is useless.